File: scaffold/src/main/java/top/xuqingquan/web/x5/MiddlewareWebClientBase.java
package top.xuqingquan.web.x5;

import com.tencent.smtt.sdk.WebViewClient;

public class MiddlewareWebClientBase extends WebViewClientDelegate {
    private MiddlewareWebClientBase mMiddleWrareWebClientBase;

    @SuppressWarnings("unused")
    public MiddlewareWebClientBase(MiddlewareWebClientBase client) {
        super(client);
        this.mMiddleWrareWebClientBase = client;
    }

    @SuppressWarnings("WeakerAccess")
    public MiddlewareWebClientBase(WebViewClient client) {
        super(client);
    }

    @SuppressWarnings("WeakerAccess")
    public MiddlewareWebClientBase() {
        super(null);
    }

    public final MiddlewareWebClientBase next() {
        return this.mMiddleWrareWebClientBase;
    }

    @Override
    public final void setDelegate(WebViewClient delegate) {
        super.setDelegate(delegate);

    }

    @SuppressWarnings("UnusedReturnValue")
    public final MiddlewareWebClientBase enq(MiddlewareWebClientBase middleWrareWebClientBase) {
        setDelegate(middleWrareWebClientBase);
        this.mMiddleWrareWebClientBase = middleWrareWebClientBase;
        return middleWrareWebClientBase;
    }

}

File: scaffold/src/main/java/top/xuqingquan/web/x5/MiddlewareWebClientBase.kt
package top.xuqingquan.web.x5

import com.tencent.smtt.sdk.WebViewClient

open class MiddlewareWebClientBase : WebViewClientDelegate {
    private var mMiddleWrareWebClientBase: MiddlewareWebClientBase? = null

    override var delegate: WebViewClient?
        get() = super.delegate
        set(delegate) {
            super.delegate = delegate

        }

    constructor(client: MiddlewareWebClientBase) : super(client) {
        this.mMiddleWrareWebClientBase = client
    }

    constructor(client: WebViewClient) : super(client)

    constructor() : super(null)

    operator fun next(): MiddlewareWebClientBase? {
        return this.mMiddleWrareWebClientBase
    }

    fun enq(middleWrareWebClientBase: MiddlewareWebClientBase): MiddlewareWebClientBase {
        delegate = middleWrareWebClientBase
        this.mMiddleWrareWebClientBase = middleWrareWebClientBase
        return middleWrareWebClientBase
    }

}
